azure-keyvault-cryptography/src/main/java/com/microsoft/azure/keyvault/cryptography/algorithms/RsaSignature.java

```diff
@@ -10,209 +10,414 @@
+            return false;
+        }
+
+        // 5. Let maskedDB be the leftmost emLen - hLen - 1 octets of EM, and
+        // let H be the next hLen octets.
+        byte[] maskedDB = Arrays.copyOfRange(EM, 0, emLen - hLen - 1);
+        byte[] H = Arrays.copyOfRange(EM, emLen - hLen - 1, emLen - 1);
+
+        // 6. If the leftmost 8emLen - emBits bits of the leftmost octet in
+        // maskedDB are not all equal to zero, output "inconsistent" and
+        // stop.
+        byte mask = 0x00;
+        for (int i = 0; i < (8 * emLen - emBits); i++) {
+            mask = (byte) (mask | (1 << i));
+            if ((maskedDB[0] & mask) != 0) {
+                return false;
+            }
+            mask = 0x00;
+        }
+
+        // 7. Let dbMask = MGF(H, emLen - hLen - 1).
+        byte[] dbMask = MGF(H, emLen - hLen - 1, messageDigest);
+
+        // 8. Let DB = maskedDB \xor dbMask.
+        byte[] DB = new byte[Math.min(maskedDB.length, dbMask.length)];
+        for (int i = 0; i < maskedDB.length; i++) {
+            DB[i] = (byte) (maskedDB[i] ^ dbMask[i]);
+        }
+
+        // 9. Set the leftmost 8emLen - emBits bits of the leftmost octet in DB
+        // to zero.
+
+        int bitMask = 0x000000FF >> (8 * emLen - emBits + 1);
+        DB[0] &= bitMask;
+
+        // 10. If the emLen - hLen - sLen - 2 leftmost octets of DB are not zero
+        // or if the octet at position emLen - hLen - sLen - 1 (the leftmost
+        // position is "position 1") does not have hexadecimal value 0x01,
+        // output "inconsistent" and stop.
+        for (int i = 0; i < (emLen - hLen - sLen - 2); i++) {
+            if (DB[i] != 0) {
+                return false;
+            }
+        }
+
+        if (DB[emLen - hLen - sLen - 2] != (byte) 0x01) {
+            return false;
+        }
+
+        // 11. Let salt be the last sLen octets of DB.
+        byte[] salt = Arrays.copyOfRange(DB, DB.length - sLen, DB.length);
+
+        // 12. Let
+        // M' = (0x)00 00 00 00 00 00 00 00 || mHash || salt ;
+        // M' is an octet string of length 8 + hLen + sLen with eight
+        // initial zero octets.
+        // 13. Let H' = Hash(M'), an octet string of length hLen.
+
+        // 14. If H = H', output "consistent." Otherwise, output "inconsistent."
+        messageDigest.update(new byte[8]);
+        messageDigest.update(mHash);
+        return Arrays.equals(H, messageDigest.digest(salt));
+    }
+    
+    /**
+     * See https://tools.ietf.org/html/rfc3447#appendix-B.2.1
+     * 
+     * @param mgfSeed seed from which mask is generated, an octet string
+     * @param maskLen intended length in octets of the mask, at most 2^32 hLen
+     * @param digest messageDigest - the hash function corresponding to the length of the hash
+     * @return mask, an octet string of length maskLen
+     */
+    protected byte[] MGF(byte[] mgfSeed, int maskLen, MessageDigest digest) {
+
+        if (maskLen > Math.pow(2, 32) * 32) {
```

Comment:
Add a comment denoting the step.  //1. If maskLen > 2^32 hLen, output "mask too long" and stop.